sourcegraph/sourcegraph
# [Exhaustive Search] Add search jobs list page

Closes https://github.com/sourcegraph/sourcegraph/issues/55708
Closes https://github.com/sourcegraph/sourcegraph/issues/55706

[Figma Designs](https://www.figma.com/file/H4ezuXmEvVeoC6UaiNSbEa/Exhaustive-search-UI?type=design&node-id=1577-12054&mode=design&t=fwULVv6bj9KuZf8D-0)

This PR adds a new page (note that this is behind `searchJobs` experimental feature flag). 
All newly added UI isn't connected to the API yet. It will be done in further PR as soon as end-to-end backend flow is in place.

## Screenshots
| Size | Desktop | Mobile |
| ----- | ------ | ------ | 
| List page with items | <img width="1115" alt="Screenshot 2023-09-05 at 22 46 05" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/0a407ad2-5acf-4517-8e90-02eb1c7fba04"> | <img width="877" alt="Screenshot 2023-09-05 at 22 46 21" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/43cfadec-cc06-4bd7-8b9f-24257347ebc2"> | 
| Dark mode | <img width="1065" alt="Screenshot 2023-09-05 at 22 47 15" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/83f07b76-dab2-42d4-9ef8-faf30d5ea712"> | <img width="857" alt="Screenshot 2023-09-05 at 23 14 55" src="https://github.com/sourcegraph/sourcegraph/assets/18492575/a411aabe-9105-486b-a9b1-660aa4c9565a"> | 

## Test plan
- Enabled `searchJobs` experimental feature flag in the settings
- Navigate to the search result page through global navigation search dropdown item
- Check general layout, note that in the webapp we don't have any mocks so there won't be any search jobs items
- Run `pnpm storybook:web` command in the sourcegraph repository root 
- Go to the `http://localhost:9001/?path=/story/web-search-jobs-searchjobspage--search-jobs-list-page` page in the browser. 
- Check layout with mocks (search jobs list page should have 5 different items) 
- Check mid-size and small screen layouts
